library/Vanilla/Embeds/SoundCloudEmbed.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: chris
+ * Date: 2018-06-05
+ * Time: 5:55 PM
+ */
+
+namespace Vanilla\Embeds;
+
+
+class SoundCloudEmbed extends Embed {
+
+    /** @inheritdoc */
+    protected $type = 'soundcloud';
+
+    /** @inheritdoc */
+    protected $domains = ['soundcloud.com'];
+
+    public function __construct() {
+        parent::__construct('soundcloud', 'image');
+    }
+
+    /**
+     * @inheritdoc
+     *
+     */
+    function matchUrl(string $url) {
+        $data = [];
+        $oembedData =[];
+        $encodedUrl = urlencode($url);
+
+        if ($this->isNetworkEnabled()) {
+            $oembedData = $this->oembed("https://soundcloud.com/oembed?url=" . $encodedUrl . "&format=json");
+            if (array_key_exists('html', $oembedData)) {
+                $data = $this->parseResponseHtml($oembedData['html']);
+            }
+        }
+
+        $data['attributes']['url'] = htmlspecialchars("https://w.soundcloud.com/player/?url=https://api.soundcloud.com/tracks/");
+
+        if (array_key_exists('height', $oembedData)) {
+            $data['height'] = $oembedData['height'];
+        }
+
+        return $data;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     */
+    public function renderData(array $data): string {
+        $height = htmlspecialchars($data['height']);
+        $track = htmlspecialchars($data['attributes']['track']);
+        $showArtwork = htmlspecialchars($data['attributes']['showArtwork']);
+        $visual = htmlspecialchars($data['attributes']['visual']);
+        $url = htmlspecialchars($data['attributes']['url']);
+
+        $result = <<<HTML
+<div class="embed embedSoundCloud">
+<iframe width="100%" height="{$height}" scrolling="no" frameborder="no" 
+    src="{$url}{$track}&show_artwork={$showArtwork}&visual={$visual}">
+</iframe>
+</div>
+HTML;
+         return $result;
+    }
+
+    /**
+     *  Parses the oembed repsonse html for permalink and other data.
+     *
+     * @param string $html
+     * @return array $data
+     */
+    public function parseResponseHtml(string $html): array {
+        $data = [];
+        preg_match('/(visual=(?<visual>true))/i', $html,$showVisual );
```

Comment:
There's some weird spacing in here. There's an extra space at the end. There's no space between the last comma and the last parameter. This also applies to the "show_artwork" parameter.